### ISSUE TITLE
F401: false positive when __all__ contains used names

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#2723]
`foo/__init__.py`:
```python
__all__ = ('foo', 'bar')
from foobarbaz import foo, bar
```

reports: ```F401: `foo` imported but unused; consider adding to `__all__` or using a redundant alias```

→ regression? latest unreleased ruff.

**Comment by charliermarsh** [comment]
Does it error if you flip the order? (I agree that either should work.)
